sql/sqlserver/server.go

```diff
@@ -123,3 +128,206 @@ func (s *Server) Send(call sqlwire.Call) {
 		},
 	}
 }
+
+// Send forwards the call for further processing.
+func (s *Server) Send(call sqlwire.Call) {
+	req := call.Args
+	resp := call.Reply
+	stmt, err := parser.Parse(req.Sql)
+	if err != nil {
+		echo(req.Sql, resp)
+		return
+	}
+	switch p := stmt.(type) {
+	case *parser.ShowColumns:
+		s.ShowColumns(p, req.Params, resp)
+	case *parser.ShowDatabases:
+		s.ShowDatabases(p, req.Params, resp)
+	case *parser.ShowIndex:
+		s.ShowIndex(p, req.Params, resp)
+	case *parser.ShowTables:
+		s.ShowTables(p, req.Params, resp)
+	case *parser.Use:
+		s.Use(p, req.Params, resp)
+	default:
+		echo(req.Sql, resp)
+	}
+}
+
+// ShowColumns of a table
+func (s *Server) ShowColumns(p *parser.ShowColumns, args []sqlwire.Datum, resp *sqlwire.Response) {
+	desc, err := s.getTableDesc(p.Table)
+	if err != nil {
+		resp.SetGoError(err)
+		return
+	}
+	rows := []sqlwire.Result_Row{}
+	for i, col := range desc.Columns {
+		t := col.Type.SQLString()
+		rows = append(rows, sqlwire.Result_Row{
+			Values: []sqlwire.Datum{
+				sqlwire.Datum{StringVal: &desc.Columns[i].Name},
```

Comment:
Pretty sure you don't need to the explicit `sqlwire.Datum`. `go fmt -s` should remove it for you.